readme.md

```diff
@@ -0,0 +1,171 @@
+
+// each associated with current logged in user in real app
+let authToken: AuthorizationToken;
+
+const router = Router();
+
+// auth flow: response with URL to redirect to Medicare.gov beneficiary login
+router.get('/authorize/authurl', bb.generateAuthorizeUrl);
+
+// auth flow: route oauth2 call back
+router.get('/bluebutton/callback', async (req: Request, res: Response) => {
+  authToken = await bb.getAuthorizationToken(authData, req.query.code, req.query.state, req.query.error);
+});
+
+// data flow
+
+// route get EOB requests
+router.get('/data/benefit', async (req: Request, res: Response) => {
+  const data = await bb.getExplanationOfBenefitData(authToken);
+  authToken = data.token;
+  res.json(data.response.data);
+});
+
+// route get Patient requests
+router.get('/data/patient', async (req: Request, res: Response) => {
+  const data = await bb.getPatientData(authToken);
+  authToken = data.token;
+  res.json(data.response.data);
+});
+
+// route get Coverage requests
+router.get('/data/coverage', async (req: Request, res: Response) => {
+  const data = await bb.getCoverageData(authToken);
+  authToken = data.token;
+  res.json(data.response.data);
+});
+
+// route get Profile requests
+router.get('/data/userprofile', async (req: Request, res: Response) => {
+  const data = await bb.getProfileData(authToken);
+  authToken = data.token;
+  res.json(data.response.data);
+});
+
+```
+
+#### Other Usage: Check Scope of Accesss Grant
+
+During authorization, the beneficiary can grant access to his/her demographic data and claims data or only claims data,
+below code snippet shows how to check the scope of the current access token:
+
+```
+import express, { Router, Request, Response } from 'express';
+
+const app = express();
+
+const bb = new BlueButton();
+const authData = bb.generateAuthData();
+
+// each associated with current logged in user in real app
+let authToken: AuthorizationToken;
+
+const router = Router();
+
+// auth flow: response with URL to redirect to Medicare.gov beneficiary login
+router.get('/authorize/authurl', bb.generateAuthorizeUrl);
+
+// auth flow: route oauth2 call back
+router.get('/bluebutton/callback', async (req: Request, res: Response) => {
+  authToken = await bb.getAuthorizationToken(authData, req.query.code, req.query.state, req.query.error);
+});
+
+const scopes: string[] = authToken.scope;
+// iterate the scopes here
+... ...
+
+1. Example of scope associated with an access token where demagraphic info allowed:
+
+scope: [
+    "patient/Coverage.read",
+    "patient/ExplanationOfBenefit.read",
+    "patient/Patient.read",
+    "profile",
+]
+
+2. Example of scope associated with an access token where demagraphic info not allowed:
+
+scope: [
+    "patient/Coverage.read",
+    "patient/ExplanationOfBenefit.read",
+]
+
+```
+
+## A Complete Sample App
+
+A Node JS React sample app can be found at:
+[CMS Blue Button Node JS Sample App](https://github.com/CMSgov/bluebutton-sample-client-python-react)
```

Comment:
Oops, this is linking out to the python sample app

Reply:
good catch, thanks!